src/category_theory/groupoid.lean

```diff
@@ -81,6 +84,10 @@ noncomputable
 def groupoid.of_is_iso (all_is_iso : ∀ {X Y : C} (f : X ⟶ Y), is_iso f) : groupoid.{v} C :=
 { inv := λ X Y f, inv f }
 
+/-- A category with a unique morphism between any two objects is a groupoid -/
+def groupoid.of_all_unique (all_unique : ∀ {X Y : C}, unique (X ⟶ Y)) :
```

Comment:
We wouldn't usually put `all` in a name. Perhaps just `groupoid.of_unique`, otherwise `groupoid.of_hom_unique`?